tests/derive/grouped_values.rs

```diff
@@ -0,0 +1,62 @@
+#![cfg(feature = "unstable-grouped")]
+use clap::Parser;
+
+#[test]
+fn test_vec_of_vec() {
+    #[derive(Parser, Debug, PartialEq)]
+    struct Opt {
+        #[arg(short = 'p', num_args = 2)]
+        points: Vec<Vec<i32>>,
```

Comment:
Technically, this is a breaking change.  Someone could have a value parser that is returning `Vec<T>` and putting that in a `Vec<_>`.

Our choices
- Put this behind `unstable-v5` flag
- Find a better way to handle introducing new type meanings (this isn't the only case, see also #3661)

Reply:
Wrapper types might be the least confusing, since they would be clearly intended for the purpose of communicating how the field should be parsed. However, they are probably the least ergonomic to use, since you would often have to unwrap the value. Although... I'm not sure how to determine for sure that the path actually points to the right type, and not a user-defined type with the same name, without require the type to be fully specified. 

To flesh out 2 some more, it could look like:

- `optional!(T)` is equivalent to the current `Option<T>`
- `values!(T)` is equivalent to the current `Vec<T>`
- `flag!()` is equivalent to the current `bool`
- `occurrences!(T)` would expand to `Vec<Vec<T>>` and opt in to getting a vec of values for each occurrence. 
- either have `optional_vec!(T)` and `optional_occurrences!(T)` or allow `optional!(values!(T))` and `optional!(occurrences!(T))`. 
- Maybe for completeness also have `required!(T)` which is the same as `T` but would make the option required?

Then again that is introducing new syntax.

Reply:
> Although... I'm not sure how to determine for sure that the path actually points to the right type, and not a user-defined type with the same name, without require the type to be fully specified.

One option would be to use deref specialization.  We could move some (or all?) of the type handling to a macro that takes in a field's type and uses deref specialization to find the right `Arg` policy type which would have a function to apply that policy to an `Arg`.

One possibility to not get derailed on this is we break thus up
- We put this PR's behavior behind the `unstable-` flag for occurrences
- We open an issue about this topic so we can raise its visibility and try to collect input.

Reply:
> One possibility to not get derailed on this is we break thus up
>
>    We put this PR's behavior behind the unstable- flag for occurrences
>     We open an issue about this topic so we can raise its visibility and try to collect input.

I'm fine with that.